pyodide_build/buildpkg.py

```diff
@@ -226,8 +226,10 @@ def run_script(buildpath: Path, srcpath: Path, pkg: Dict[str, Any]):
     finally:
         os.chdir(orig_path)
 
-    with open(buildpath / ".packaged", "wb") as fd:
-        fd.write(b"\n")
+    # If library, we're done so create .packaged file
+    if pkg["build"].get("library"):
+        with open(buildpath / ".packaged", "wb") as fd:
+            fd.write(b"\n")
```

Comment:
Would you mind adding a short test to, `pyodide_build/tests/test_buildpkg.py`? With something like,
```py
from textwrap import dedent 

@pytest.mark.parametrize('is_library', [True, False])
def test_run_script(is_library, tmpdir):
    build_dir = tmpdir.mkdir('build')
    src_dir = tmpdir.mkdir('build')
    script = dedent('''
         touch out.txt
    """)
    pkg = {'build': {'script': script}}
    run_script(build_dir, src_dir, pkg)
    assert (build_dir / 'out.txt').exists()
    if library:
        assert (build_dir / '.packaged').exists()
    else:
        assert not (build_dir / '.packaged').exists()
```
(Paths and some details might need adjusting, I haven't tested it). I realize we should have done this when initially adding this feature, but now is also good.